enumerable_methods.rb

```diff
@@ -0,0 +1,65 @@
+module Enumerable
+  def my_each
+    return unless block_given?
+
+    for i in self
+      yield i
+    end
+  end
+
+  def my_each_with_index
+    return unless block_given?
+
+    index = 0
+    for i in self
+      yield i, index
+      index += 1
+    end
+  end
+
+  def my_select
+    arr = []
+    my_each { |n| arr.push(n) if yield n } if block_given?
+    arr
+  end
+
+  def my_all?
+    my_select { |n| yield n }.length == length if block_given?
+  end
+
+  def my_any?
+    my_select { |n| yield n }.length.positive? if block_given?
+  end
+
+  def my_none?
+    !my_any? if block_given?
+  end
```

Comment:
- [ ] enumerables #my_none? returns false if the block ever returns true for all elements
      Failure/Error: expect(array.my_none?(&false_block)).to eq(array.none?(&false_block))

- [ ] enumerables #my_none? returns false if the block ever returns true for all elements::range
      Failure/Error: expect(range.my_none?(&false_block)).to eq(range.none?(&false_block))

- [ ] enumerables #my_none? when no block or argument is given returns true only if none of the collection members is true
      Failure/Error: expect(false_array.my_none?).to be true

- [ ] enumerables #my_none? when no block or argument is given returns false only if one of the collection members is true
      Failure/Error: expect(true_array.my_none?).to be false

- [ ] enumerables #my_none? when a class is passed as an argument returns true if none of the collection is a member of such class
      Failure/Error: expect(array.my_none?(String)).to be array.my_none?(String)

- [ ] enumerables #my_none? when a class is passed as an argument returns true if none of the collection is a member of such class::Numeric
      Failure/Error: expect(array.my_none?(Numeric)).to be array.my_none?(Numeric)

- [ ] enumerables #my_none? when a class is passed as an argument returns false if any of the collection is a member of such class
      Failure/Error: expect(array.my_none?(String)).to be array.my_none?(String)

- [ ] enumerables #my_none? when a Regex is passed as an argument returns true if none of the collection matches the Regex
      Failure/Error: expect(words.my_none?(/z/)).to be words.none?(/z/)

- [ ] enumerables #my_none? when a Regex is passed as an argument returns false if any of the collection matches the Regex
      Failure/Error: expect(words.my_none?(/d/)).to be words.none?(/d/)

- [ ] enumerables #my_none? when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
      Failure/Error: expect(words.my_none?(5)).to be words.none?(5)

- [ ] enumerables #my_none? when a pattern other than Regex or a Class is given returns false only if one of the collection matches the pattern
      Failure/Error: expect(words.my_none?(5)).to be words.none?(5)

- [ ]